### PR TITLE
Created a github action to build synapse package. Also deleted Onion.dmg file. 

### PR DESCRIPTION
**Pull request by alexjyong**
Went and whipped up a github action to build synapse wrapper.  Built out and image, pushed it to my embassy, and it seems to be have installed and ran ok. This will allow you to use github actions to build a package without having to set up a build environment on your local machine. In theory, you should be able to use this framework for other services if you just switch out synapse for whatever.

Also updated the Readme because I found a missing step.

One limitation is that I can't run this in https://github.com/nektos/act locally, which would had been super useful. The rust action at the beginning breaks. Not too sure if it's a bug with my local machine, or with `act` itself. I opened up an issue with them, https://github.com/nektos/act/issues/1295 

Side note, I did update the manifest.yaml temporarily to have the name and id be synapse2 temporarily, but that only seemed to overwrite my current install, and not deploy out a separate service.  

Also deleted ElementOnion.dmg.

### CODE REVIEW

**Comment by alexjyong** [comment]
Ok, currently have an issue with getting the package to run on matrix unfortunately.

I have reached out to @chrisguida and @k0gen about this. Might be something wrong with the build process. 

I am also looking into some tweaks for the current process as well, found a few mistakes in the repo checkouts.

File: .github/workflows/buildService.yml
Comment (chrisguida) [inline .github/workflows/buildService.yml:50]: TOML is actually not used, this can be removed

**Comment by chrisguida** [comment]
Can you provide a link to the artifact produced by this build?

**Comment by alexjyong** [comment]
@chrisguida sent you this in the other channel, but I'll post it here as well. 

Last successful build. I believe it was done with the update branch as well: https://github.com/alexjyong/synapse-wrapper/actions/runs/2821775057

This was the github action file used: https://github.com/alexjyong/synapse-wrapper/blob/d39d8e6655de0f767ebea352ab28d0be5dc04ef1/.github/workflows/buildService.yml



Now, the most recent build at the time of writing is this:
https://github.com/alexjyong/synapse-wrapper/runs/7758452172?check_suite_focus=true

Failed with this:

```
embassy-sdk pack
embassy-sdk verify s9pk synapsealt.s9pk
S9PK Validation Error: Contains image for incorrect package: id synapse
make: *** [Makefile:18: verify] Error 24
```
Using this github action code: https://github.com/alexjyong/synapse-wrapper/blob/5a5fe49d4c9b7d8f7aaead27aee34544d929f6c2/.github/workflows/buildService.yml

**Comment by alexjyong** [comment]
@chrisguida here is the link to the last successful artifact (at the time of writing)
https://github.com/alexjyong/synapse-wrapper/suites/7724169508/artifacts/323636292

**Comment by alexjyong** [comment]
@chrisguida https://github.com/alexjyong/synapse-wrapper/suites/7755197823/artifacts/325827390 

Newer artifact. No ID changing on this one FWIW.

Code used:
https://github.com/alexjyong/synapse-wrapper/blob/87ae72098b4c5973042cdc6b84d31f8109fcf074/.github/workflows/buildService.yml

**Comment by chrisguida** [comment]
Is there a way to just zip the s9pk? Don't see why we need all these nested directories:

Screen Shot 2022-08-10 at 2 29 34 PM

**Comment by alexjyong** [comment]
@chrisguida  Yeah, there should be a way. I think that came up from the way I was grabbing the file. I currently have a build job going that might fix it.

**Comment by alexjyong** [comment]
https://github.com/alexjyong/synapse-wrapper/actions/runs/2835226552

@chrisguida fixed the zip imports issue. 

When will the update branch be merged in? Right now the code is pulling it specifically.

**Comment by chrisguida** [comment]
It can be merged once @k0gen says it's done, and it gets an approving review

**Comment by alexjyong** [comment]
@chrisguida right on. Any other feedback with this PR?

**Comment by chrisguida** [comment]
Not at the moment, it looks great, thanks for putting this together! 

We should probably update it to point at master before merging.

**Comment by chrisguida** [comment]
And since the build will fail, we should probably hold off on doing that until @k0gen's PR is merged.

**Comment by alexjyong** [comment]
@chrisguida is master fixed though? I recall there was build issues with master.

**Comment by alexjyong** [comment]
Merged conflicts fixed at the time of writing. Waiting for build to finish.

**Comment by alexjyong** [comment]
Successful build https://github.com/alexjyong/synapse-wrapper/runs/7788858117?check_suite_focus=true

@chrisguida if you wanna do a final review

**Comment by alexjyong** [comment]
Also deleted ElementOnion.dmg @k0gen

**Comment by chrisguida** [comment]
The artifact is still 1.42

Screen Shot 2022-08-11 at 11 10 16 AM

**Review by chrisguida — CHANGES_REQUESTED** [review]
Please update the title or description with the deletion of the .dmg file

**Comment by chrisguida** [comment]
@k0gen when you merge this, please please please squash. I don't want 68 commits for this in master

**Comment by chrisguida** [comment]
@alexjyong If you want to rebase this on master we can test. Otherwise I'm fine with YOLOing this into the S9 fork and fixing after. But the first way is cleaner if you're not 100% sure it will work.

**Comment by k0gen** [comment]
ok the `master` is officially frozen

**Review by chrisguida — CHANGES_REQUESTED** [review]
@alexjyong Can you rebase this on master? I promise this is the last time. It's just that there's a bunch of commits now that shouldn't be here

**Comment by chrisguida** [comment]
@alexjyong Is this PR still needed for anything?

**Comment by alexjyong** [comment]
@chrisguida it is not. I kept it around until the other one was merged in JIC. Closing this out now.

**Pull request closed without merging**
